modules/installation-creating-upi-worker-machineset.adoc

```diff
@@ -0,0 +1,144 @@
+
+.Procedure
+
+. Optionally, launch worker nodes that are controlled by the machine API.
+. View the list of MachineSets in the `openshift-machine-api` namespace:
++
+----
+$ oc get machinesets --namespace openshift-machine-api
+NAME                           DESIRED   CURRENT   READY     AVAILABLE   AGE
+test-tkh7l-worker-us-east-2a   1         1                               11m
+test-tkh7l-worker-us-east-2b   1         1                               11m
+test-tkh7l-worker-us-east-2c   1         1                               11m
+----
++
+Note the `NAME` of each MachineSet. Because you use a different subnet than the
+installation program expects, the worker MachineSets do not use the correct
+network settings. You must edit each of these MachineSets.
+
+. Edit each worker MachineSet to provide the correct values for your cluster:
++
+----
+$ oc edit machineset --namespace openshift-machine-api test-tkh7l-worker-us-east-2a -o yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  creationTimestamp: 2019-03-14T14:03:03Z
+  generation: 1
+  labels:
+    machine.openshift.io/cluster-api-cluster: test-tkh7l
+    machine.openshift.io/cluster-api-machine-role: worker
+    machine.openshift.io/cluster-api-machine-type: worker
+  name: test-tkh7l-worker-us-east-2a
+  namespace: openshift-machine-api
+  resourceVersion: "2350"
+  selfLink: /apis/machine.openshift.io/v1beta1/namespaces/openshift-machine-api/machinesets/test-tkh7l-worker-us-east-2a
+  uid: e2a6c8a6-4661-11e9-a9b0-0296069fd3a2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: test-tkh7l
+      machine.openshift.io/cluster-api-machineset: test-tkh7l-worker-us-east-2a
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: test-tkh7l
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: test-tkh7l-worker-us-east-2a
+    spec:
+      metadata:
+        creationTimestamp: null
+      providerSpec:
+        value:
+          ami:
+            id: ami-07e0e0e0035b5a3fe <1>
+          apiVersion: awsproviderconfig.openshift.io/v1beta1
+          blockDevices:
+          - ebs:
+              iops: 0
+              volumeSize: 120
+              volumeType: gp2
+          credentialsSecret:
+            name: aws-cloud-credentials
+          deviceIndex: 0
+          iamInstanceProfile:
+            id: test-tkh7l-worker-profile
+          instanceType: m4.large
+          kind: AWSMachineProviderConfig
+          metadata:
+            creationTimestamp: null
+          placement:
+            availabilityZone: us-east-2a
+            region: us-east-2
+          publicIp: null
+          securityGroups:
+          - filters:
+            - name: tag:Name
+              values:
+              -  test-tkh7l-worker-sg <2>
```

Comment:
@wking, do lines 95 and 117 look right to you?

Reply:
yup, `{infraName}-{role}-{sg}` matches [this][1].

[1]: https://github.com/openshift/installer/blob/6e8d74889800a841a6ed197c6ec62ea08734fdea/upi/aws/cloudformation/03_cluster_security.yaml#L52-L76

Reply:
Are your sure here? Per my test result, https://bugzilla.redhat.com/show_bug.cgi?id=1697968#c11, security group would be created with '{role}-{sg}-{infraName}' as 'GroupName' and empty 'Name'. While machine api is expecting '{infraName}-{role}-{sg}' as 'Name'. Do I miss something?